lib/src/process_pool.dart

```diff
@@ -98,9 +164,65 @@ class WorkerJob {
   Exception? exception;
 
   @override
-  String toString() {
-    return command.join(' ');
+  void addToQueue(List<DependentJob> jobs) {
+    jobs.add(this);
   }
+
+  @override
+  String toString() => name;
+}
+
+/// A job that groups other jobs.
+///
+/// Each worker job, except the first, are made to be dependent upon the
+/// previous worker job.
+///
+/// This group job is automatically dependent upon all of the [workers] it manages.
```

Comment:
some of the doc here is more of implementation details, maybe just say this job finishes when all the worker finish.

Reply:
Done.